website/content/docs/api-gateway/upgrades.mdx

```diff
@@ -9,6 +9,151 @@ description: >-
 
 This topic describes how to upgrade Consul API Gateway.
 
+## Upgrade to v0.4.0
+
+Consul API Gateway v0.4.0 introduces support for [Gateway API v0.5.0](https://github.com/kubernetes-sigs/gateway-api/releases/tag/v0.5.0), the graduated v1beta1 `GatewayClass`, `Gateway` and `HTTPRoute` resources, and the new [`ReferenceGrant`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferenceGrant) resource, which is intended to replace the identical [`ReferencePolicy`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io/v1alpha2.ReferencePolicy). This release includes backwards-compatibiltiy support for existing `ReferencePolicy` resources, but this will be removed in a future release, so we encourage users to migrate to `ReferenceGrant` after upgrading.
+
+### Requirements
+
+Ensure that the following requirements are met prior to upgrading:
+
+- Consul API Gateway should be running version v0.3.0.
+
+### Procedure
+
+1. Proceed with the [standard upgrade](#standard-upgrade), then after you've completed the upgrade, return to the [post-upgrade configuration changes instructions](#v0.4.0-post-upgrade-configuration-changes) to update your `GatewayClass`, `Gateway` and `HTTPRoute` resources from v1alpha2 to v1beta1, and replace your `ReferencePolicy` resources with `ReferenceGrant`.
```

Comment:
Looks like this suggestion conflicts with a different one I just merged that similarly splits into two steps, but uses the terms `complete` instead of `perform` and `instructions` instead of `procedure` - could you resolve this and open a new suggestion against the current state?